plugins/ccgx/fu-ccgx-hpi-common.h

```diff
@@ -0,0 +1,344 @@
+	CY_PD_RESP_FLASH_DATA_AVAILABLE,
+	CY_PD_RESP_INVALID_COMMAND		= 0x05,
+	CY_PD_RESP_COLLISION_DETECTED,
+	CY_PD_RESP_FLASH_UPDATE_FAILED,
+	CY_PD_RESP_INVALID_FW,
+	CY_PD_RESP_INVALID_ARGUMENTS,
+	CY_PD_RESP_NOT_SUPPORTED,
+	CY_PD_RESP_TRANSACTION_FAILED		= 0x0C,
+	CY_PD_RESP_PD_COMMAND_FAILED,
+	CY_PD_RESP_UNDEFINED,
+	CY_PD_RESP_RA_DETECT			= 0x10,
+	CY_PD_RESP_RA_REMOVED,
+
+	/* device specific events */
+	CY_PD_RESP_RESET_COMPLETE		= 0x80,
+	CY_PD_RESP_MESSAGE_QUEUE_OVERFLOW,
+
+	/* type-c specific events */
+	CY_PD_RESP_OVER_CURRENT_DETECTED,
+	CY_PD_RESP_OVER_VOLTAGE_DETECTED,
+	CY_PD_RESP_TYPC_C_CONNECTED,
+	CY_PD_RESP_TYPE_C_DISCONNECTED,
+
+	/* pd specific events and asynchronous messages */
+	CY_PD_RESP_PD_CONTRACT_ESTABLISHED,
+	CY_PD_RESP_DR_SWAP,
+	CY_PD_RESP_PR_SWAP,
+	CY_PD_RESP_VCON_SWAP,
+	CY_PD_RESP_PS_RDY,
+	CY_PD_RESP_GOTOMIN,
+	CY_PD_RESP_ACCEPT_MESSAGE,
+	CY_PD_RESP_REJECT_MESSAGE,
+	CY_PD_RESP_WAIT_MESSAGE,
+	CY_PD_RESP_HARD_RESET,
+	CY_PD_RESP_VDM_RECEIVED,
+	CY_PD_RESP_SRC_CAP_RCVD,
+	CY_PD_RESP_SINK_CAP_RCVD,
+	CY_PD_RESP_DP_ALTERNATE_MODE,
+	CY_PD_RESP_DP_DEVICE_CONNECTED,
+	CY_PD_RESP_DP_DEVICE_NOT_CONNECTED,
+	CY_PD_RESP_DP_SID_NOT_FOUND,
+	CY_PD_RESP_MULTIPLE_SVID_DISCOVERED,
+	CY_PD_RESP_DP_FUNCTION_NOT_SUPPORTED,
+	CY_PD_RESP_DP_PORT_CONFIG_NOT_SUPPORTED,
+	CY_PD_HARD_RESET_SENT,
+	CY_PD_SOFT_RESET_SENT,
+	CY_PD_CABLE_RESET_SENT,
+	CY_PD_SOURCE_DISBALED_STATE_ENTERED,
+	CY_PD_SENDER_RESPONSE_TIMER_TIMEOUT,
+	CY_PD_NO_VDM_RESPONSE_RECEIVED
+} CyPDResp;
+
+typedef enum {
+	HPI_RESPONSE_NO_RESPONSE,
+	HPI_RESPONSE_SUCCESS			= 0x02,
+	HPI_RESPONSE_FLASH_DATA_AVAILABLE,
+	HPI_RESPONSE_INVALID_COMMAND		= 0x05,
+	HPI_RESPONSE_FLASH_UPDATE_FAILED	= 0x07,
+	HPI_RESPONSE_INVALID_FW,
+	HPI_RESPONSE_INVALID_ARGUMENT,
+	HPI_RESPONSE_NOT_SUPPORTED,
+	HPI_RESPONSE_PD_TRANSACTION_FAILED	= 0x0C,
+	HPI_RESPONSE_PD_COMMAND_FAILED,
+	HPI_RESPONSE_UNDEFINED_ERROR		= 0x0F,
+	HPI_EVENT_RESET_COMPLETE		= 0x80,
+	HPI_EVENT_MSG_OVERFLOW,
+	HPI_EVENT_OC_DETECT,
+	HPI_EVENT_OV_DETECT,
+	HPI_EVENT_CONNECT_DETECT,
+	HPI_EVENT_DISCONNECT_DETECT,
+	HPI_EVENT_NEGOTIATION_COMPLETE,
+	HPI_EVENT_SWAP_COMPLETE,
+	HPI_EVENT_PS_RDY_RECEIVED		= 0x8A,
+	HPI_EVENT_GOTO_MIN_RECEIVED,
+	HPI_EVENT_ACCEPT_RECEIVED,
+	HPI_EVENT_REJECT_RECEIVED,
+	HPI_EVENT_WAIT_RECEIVED,
+	HPI_EVENT_HARD_RESET_RECEIVED,
+	HPI_EVENT_VDM_RECEIVED			= 0x90,
+	HPI_EVENT_SOURCE_CAP_RECEIVED,
+	HPI_EVENT_SINK_CAP_RECEIVED,
+	HPI_EVENT_DP_MODE_ENTERED,
+	HPI_EVENT_DP_STATUS_UPDATE,
+	HPI_EVENT_DP_SID_NOT_FOUND		= 0x96,
+	HPI_EVENT_DP_MANY_SID_FOUND,
+	HPI_EVENT_DP_NO_CABLE_SUPPORT,
+	HPI_EVENT_DP_NO_UFP_SUPPORT,
+	HPI_EVENT_HARD_RESET_SENT,
+	HPI_EVENT_SOFT_RESET_SENT,
+	HPI_EVENT_CABLE_RESET_SENT,
+	HPI_EVENT_SOURCE_DISABLED,
+	HPI_EVENT_SENDER_TIMEOUT,
+	HPI_EVENT_VDM_NO_RESPONSE,
+	HPI_EVENT_UNEXPECTED_VOLTAGE,
+	HPI_EVENT_ERROR_RECOVERY,
+	HPI_EVENT_EMCA_DETECT			= 0xA6,
+	HPI_EVENT_RP_CHANGE_DETECT		= 0xAA,
+	HPI_EVENT_TB_ENTERED			= 0xB0,
+	HPI_EVENT_TB_EXITED
+} HPIResp;
```

Comment:
With the big gaps in documented events, I would question whether this being an enum is even valuable.  Should it just be a bunch of #defines for the ones documented?

Reply:
I was torn about this; on one hand it's very verbose, on the other hand you can diff from the Windows sources too.

Reply:
Ah if it's like this in the Windows sources and more easily diffable for changes, it's better to keep it this way I'd say.  New events that are otherwise accidentally forgotten to add will stand out much more easily.